### DISCUSSION TITLE
General Item Structure

### DISCUSSION THREAD

**Discussion by darrenwiens** [stac-extensions/video#1] [General]
Video properties are currently placed in `item:properties`, meaning these properties will apply to the entire item, and also implies that an item will only contain a single video, which makes sense to me because each video should have a unique spatiotemporal extent.

Is it possible for a single item to contain multiple videos with the same spatiotemporal extent (e.g. WAMI)? If so, these properties could be nested under each `video` asset.

```
{
...
    "properties": {
        ...
        "video:frame_rate": 29.97002997002997,
        "video:frame_count": 4441,
        "video:motion_type": "full",
        "video:shape": [
          1080,
          1920
        ],
        "video:codec_name": "h264"
    }
}
```

**Comment by m-mohr** [comment]
You may want to look into asset roles and use them instead of the asset keys. This way you can also have more than one video per item (although not sure that's a good idea).

**Reply by darrenwiens** [reply]
Thanks, this is a good idea.

So, for an item containing assets for video, frame-centers, and frame-geometries, there would be arbitrarily named assets, with roles: ["video"], ["frame_centers"], and ["frame_geometries"].

Is there any convention for grouping assets, either through a common role or other property? For example, group the assets associated with one video, and a different group for those associated with a second video?

**Reply by darrenwiens** [reply]
Addressed in PR: https://github.com/stac-extensions/video/pull/15

**Reply by m-mohr** [reply]
Great. I think it makes a lot of sense to also use the role for grouping rather than the key. Actually thinking to implement role groups in STAC Browser now... 😊 So you'd probably have 3 roles...

**Reply by darrenwiens** [reply]
Sure, I think that's better, too. How will STAC Browser know which role to group by, or will that require a new extension? 😄

**Reply by m-mohr** [reply]
Well, that's the question I or we need to solve. Especially with multiple roles.

You could group: metadata, data, other.
Or you may want to group by video when using this extension.

Well... maybe it's a can of worms?

Still, roles is the place to group, not the asset key IMHO.

**Reply by darrenwiens** [reply]
Addressed in PR: https://github.com/stac-extensions/video/pull/16

Basically, a recommendation for an asset role like `group:groupname`, which I believe could be a future extension (`asset-group`?)
